tools/storage_mon.c

```diff
@@ -72,65 +77,58 @@ static int open_device(const char *device, int verbose)
 	res = lseek(device_fd, seek_spot, SEEK_SET);
 	if (res < 0) {
 		fprintf(stderr, "Failed to seek %s: %s\n", device, strerror(errno));
-		close(device_fd);
-		return -1;
+		goto error;
 	}
 	if (verbose) {
 		printf("%s: reading from pos %ld\n", device, seek_spot);
 	}
-	return device_fd;
-}
-
-/* Check one device */
-static void *test_device(const char *device, int verbose, int inject_error_percent)
-{
-	int device_fd;
-	int sec_size = 0;
-	int res;
-	void *buffer;
-
-	if (verbose) {
-		printf("Testing device %s\n", device);
-	}
 
-	device_fd = open_device(device, verbose);
-	if (device_fd < 0) {
-		exit(-1);
-	}
+	if (flags & O_DIRECT) {
+		int sec_size = 0;
+		void *buffer;
 
 #ifdef __FreeBSD__
-	ioctl(device_fd, DIOCGSECTORSIZE, &sec_size);
+		res = ioctl(device_fd, DIOCGSECTORSIZE, &sec_size);
 #else
-	ioctl(device_fd, BLKSSZGET, &sec_size);
+		res = ioctl(device_fd, BLKSSZGET, &sec_size);
 #endif
-	if (sec_size == 0) {
-		fprintf(stderr, "Failed to stat %s: %s\n", device, strerror(errno));
-		goto error;
-	}
+		if (res < 0) {
+			fprintf(stderr, "Failed to stat %s: %s\n", device, strerror(errno));
```

Comment:
why don't we simply say that getting the block-size did fail?

Reply:
I agree. https://github.com/ClusterLabs/resource-agents/pull/1797/commits/db97e055a17526cec056c595844a9d8851e3ee19